docs/user_guide/02_medrecord_intro.md

```diff
@@ -0,0 +1,239 @@
+# MedRecord
+
+## Preface
+
+Just like every major library has its core class — [PyTorch](https://pytorch.org/) has `torch.Tensor`, [Numpy](https://numpy.org/) has `np.array` — MedModels revolves around `mm.MedRecord` as its central object.
+
+MedModels delivers advanced data analytics methods out-of-the-box by utilizing a structured approach to data storage. This is enabled by the MedRecord class, which organizes data of any complexity within a graph structure. With its Rust backend implementation, MedRecord guarantees high performance, even when working with extremely large datasets.
+
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 3
+---
+```
+
+## Adding Nodes to a MedRecord
+
+Let's begin by introducing some sample medical data:
+
+:::{list-table} Patients
+:widths: 15 15 15 15
+:header-rows: 1
+
+*   - ID
+    - Age
+    - Sex
+    - Loc
+*   - Patient 01
+    - 72
+    - M
+    - USA
+*   - Patient 02
+    - 74
+    - M
+    - USA
+*   - Patient 03
+    - 64
+    - F
+    - GER
+:::
+
+This data, stored for example in a Pandas DataFrame, looks like this:
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 5-13
+---
+```
+
+In the example below, we create a new MedRecord using the builder pattern. We instantiate a `MedRecordBuilder` and instruct it to add the Pandas DataFrame as nodes, using the 'ID' column for indexing. Additionally, we assign these nodes to the group 'Patients'.
+The Builder Pattern simplifies creating complex objects by constructing them step by step. It improves flexibility, readability, and consistency, making it easier to manage and configure objects in a controlled way.
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 43
+---
+```
+
+The MedModels MedRecord object, `record`, now contains three patients. Each patient is identified by a unique index and has specific attributes, such as age, sex, and location. These patients serve as the initial nodes in the graph structure of our MedRecord, and are represented as follows:
+
+```{image} https://raw.githubusercontent.com/limebit/medmodels-static/main/imgs/user_guide/02/02_medrecord_intro_01.png
+:class: transparent-image
+```
+We can now proceed by adding additional data, such as the following medications.
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 15-18
+---
+```
+Using the builder pattern to construct the MedRecord allows us to pass as many nodes and edges as needed. If nodes are not added during the initial graph construction, they can easily be added later to an existing MedRecord by calling `add_nodes`, where you provide the DataFrame and specify the column containing the node indices.
+
+```{literalinclude} scripts/02_medrecord_intro.py
+---
+language: python
+lines: 46
```

Comment:
This python line does not add a "Medications" group, as portrayed in the image. It is misleading. Either we establish how to add groups now or afterwards. Maybe not add any groups at all until referring to them in section 4.

Reply:
This will be fixed with an update that is to be discussed today: The add_nodes function will get an extra argument to pass a group directly, then I will change this code.